arbitrator/wasm-libraries/user-host-trait/src/lib.rs

```diff
@@ -477,6 +477,27 @@ pub trait UserHost: GasMeteredMachine {
         trace!("account_codehash", self, address, hash)
     }
 
+    fn account_code(&mut self, address: u32, dest: u32, size: u32) -> Result<(), Self::Err> {
+        self.buy_ink(HOSTIO_INK + EVM_API_INK)?;
+        let address = self.read_bytes20(address)?;
+
+        let (code, gas_cost) = self.evm_api().account_code(address);
+        assert!(code.len() <= size as usize);
```

Comment:
I copied this approach from `read_return_data` - is there a reason the two situations are different, or do we need to change the API there to?

Reply:
Also, do you just mean to write `min(code.len(), size)` bytes? Not sure why `offset` is involved.

Reply:
In our case we pass in the `size` to `read_return_data`, which Go uses to limit the return data to no more than `size`. The assert just helps with testing. It's ok to leave it in here, but only if we make sure the Go side limits the amount of data we return.

For the offset part, the idea is we'll want to mimic the EVM's behavior, which allows users to select an offset into the code from which to read. This lets the smart contract dev be strategic about which bytes they want, saving gas. Since we're going to want this API to include this capability, we'll need to do something like we do for read_return_data (I suggest matching the EVM, maybe minus the zero-filling part for the `CODE_COPY` instruction)